src/dynamic_reconfigure/parameter_generator_catkin.py

```diff
@@ -575,6 +575,27 @@ def write_params(group):
 #        print >> f, self.msgname, "config", "# What the node's configuration was actually set to."
 #        f.close()
     
+    def _rreplace_str_with_val_in_dict(self, orig_dict, old_str, new_val):
+        # Recursively replace any match of old_str by new_val in a dictionary
+        for k, v in orig_dict.items():
+            if isinstance(v, dict):
+                self._rreplace_str_with_val_in_dict(v, old_str, new_val)
+            elif isinstance(v, list):
+                for i in v:
+                    if isinstance(i, str) and i == old_str:
+                        i = new_val
```

Comment:
It would be clearer (in my opinion) if you assigned it directly back into `orig_dict` as you do in line 590.